Formula/ghostunnel.rb

```diff
@@ -0,0 +1,26 @@
+class Ghostunnel < Formula
+  desc "Simple SSL/TLS proxy with mutual authentication"
+  homepage "https://github.com/ghostunnel/ghostunnel"
+  url "https://github.com/ghostunnel/ghostunnel/archive/refs/tags/v1.5.3.tar.gz"
+  sha256 "008b96d507b80480f4e78d31bdfbb37b794f51c2a7a80a8f3561844e2739b5a5"
+  license "Apache-2.0"
+  head "https://github.com/ghostunnel/ghostunnel.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}")
+  end
+
+  test do
+    server = fork do
+      exec bin/"ghostunnel", "client", "--listen=localhost:8888", "--target=localhost:80",
+        "--disable-authentication", "--shutdown-timeout=1s", "--connect-timeout=1s"
+    end
+    sleep 1
+    shell_output("curl -o /dev/null http://localhost:8888/", 56)
+  ensure
+    Process.kill "KILL", server
+    Process.wait server
+  end
```

Comment:
Brew will kill any tests on its own, and picking a generated free port makes testing easier.

I'm just wondering if it would be better to fetch brew.sh instead of assuming something on localhost.
```suggestion
  test do
    port = free_port
    server = fork do
      exec bin/"ghostunnel", "client", "--listen=localhost:#{port}", "--target=localhost:80",
        "--disable-authentication", "--shutdown-timeout=1s", "--connect-timeout=1s"
    end
    sleep 1
    shell_output("curl -o /dev/null http://localhost:#{port}/", 56)
  end
```

Reply:
`free_port` is what I wanted.  I switched to port 4 (reserved) to test that rejected connections propagate correctly.